src/Template/Bake/Template/index.ctp

```diff
@@ -16,6 +16,7 @@
 <?php
 /**
   * @var \<%= $namespace %>\View\AppView $this
+  * @var \<%= $namespace %>\Model\Entity\<%= $entityClass %>[] $<%= $pluralVar %>
```

Comment:
Maybe - like my ide helper - it should check if those entities actually exist, and if not fall back to some interface for entities or collections.
In this case you have still at least some annotations, even if you do not want to use concrete entities.